src/bandersnatch/tests/test_verify.py

```diff
@@ -208,5 +212,90 @@ async def test_metadata_verify(monkeypatch: MonkeyPatch) -> None:
     await metadata_verify(fc, fa)  # type: ignore
 
 
+@pytest.mark.asyncio
+async def test_get_latest_json_timeout(tmp_path: Path) -> None:
+    class FakeArgs:
+        delete = True
+        dry_run = False
+        json_update = True
+        workers = 2
+
+    fa = FakeArgs()
+    fc = FakeConfig()
+
+    master = Master(fc.get("mirror", "master"))
+    url_fetch_timeout = mock.AsyncMock(side_effect=ServerTimeoutError)  # type: ignore
+    master.url_fetch = url_fetch_timeout  # type: ignore
+
+    jsonpath = tmp_path / "web" / "json"
+    jsonpath.mkdir(parents=True)
+    jsonfile = jsonpath / "bandersnatch"
+    jsonfile.touch()
+    all_package_files: List[str] = []
+
+    await verify(master, fc, "bandersnatch", tmp_path, all_package_files, fa)  # type: ignore # noqa: E501
+    assert jsonfile.exists()
+    assert not all_package_files
+
+
+@pytest.mark.asyncio
+async def test_get_latest_json_404(tmp_path: Path) -> None:
+    class FakeArgs:
+        delete = True
+        dry_run = False
+        json_update = True
+        workers = 2
+
+    fa = FakeArgs()
+    fc = FakeConfig()
+
+    master = Master(fc.get("mirror", "master"))
+    url_fetch_404 = mock.AsyncMock(  # type: ignore
```

Comment:
I think typing actually helped here - AsyncMock is not available to 3.8 and we still support 3.6.1 ... Maybe now could be the time to go to >= 3.8 ...
```suggestion
    url_fetch_404 = mock.AsyncMock(  # type: ignore
```

Reply:
Sorry. Happy to change if not ready for >= 3.8. Will await decision.

Reply:
This change would affect, for example LTS users of Ubutnu 18.04 and 16.04 wouldn't it?

Reply:
Yeah, Ubuntu 18.04 LTS only has CPython 3.6.something pre-installed and Ubuntu 16.04 LTS only has CPython 3.5.something pre-installed. (I know CPython 2.7.something is also included on both but let's just ignore them)

Reply:
bandersnatch has ignored python2 since I took over :D My first contributions were moving from Python2 only to Python3 :)